pallets/parachain-staking/src/tests.rs

```diff
@@ -3926,3 +3836,504 @@ fn update_total_stake_no_collator_changes() {
+		.build()
+		.execute_with(|| {
+			// claiming or incrementing should not be possible with zero counter
+			assert_noop!(
+				StakePallet::increment_collator_rewards(Origin::signed(1)),
+				Error::<Test>::RewardsNotFound
+			);
+			assert_noop!(
+				StakePallet::increment_delegator_rewards(Origin::signed(2)),
+				Error::<Test>::RewardsNotFound
+			);
+			(1..=3).for_each(|id| {
+				assert_noop!(
+					StakePallet::claim_rewards(Origin::signed(id)),
+					Error::<Test>::RewardsNotFound,
+				);
+			});
+
+			// note once to set counter to 1
+			StakePallet::note_author(1);
+			assert_eq!(StakePallet::reward_count(1), 1);
+			assert!(StakePallet::reward_count(2).is_zero());
+
+			// claiming should not be possible before incrementing rewards
+			(1..=3).for_each(|id| {
+				assert_noop!(
+					StakePallet::claim_rewards(Origin::signed(id)),
+					Error::<Test>::RewardsNotFound
+				);
+			});
+
+			// increment rewards for 2 and match counter to collator
+			assert_ok!(StakePallet::increment_delegator_rewards(Origin::signed(2)));
+			assert_eq!(StakePallet::reward_count(2), 1);
+			let rewards_2 = StakePallet::rewards(2);
+			assert!(!rewards_2.is_zero());
+			assert!(StakePallet::reward_count(3).is_zero());
+			assert!(StakePallet::rewards(3).is_zero());
+
+			// should set rewards for delegator 3 as well
+			assert_ok!(StakePallet::increment_collator_rewards(Origin::signed(1)));
+			assert!(StakePallet::reward_count(1).is_zero());
+			assert!(!StakePallet::rewards(1).is_zero());
+			// counter of delegators should be reset to 0 (= cols counter)
+			assert!(StakePallet::reward_count(2).is_zero());
+			assert!(StakePallet::reward_count(3).is_zero());
+			// rewards of 2 should not be changed
+			assert_eq!(StakePallet::rewards(2), rewards_2);
+			// 3 should have rewards now (passively)
+			assert!(!StakePallet::rewards(3).is_zero());
+
+			// claim for 1 to move rewards into balance
+			assert_ok!(StakePallet::claim_rewards(Origin::signed(1)));
+			assert!(StakePallet::reward_count(1).is_zero());
+			assert!(StakePallet::rewards(1).is_zero());
+			// delegator situation should be unchanged
+			assert!(Balances::free_balance(&1) > DECIMALS);
+			assert_eq!(Balances::free_balance(&2), DECIMALS);
+			assert_eq!(Balances::free_balance(&3), DECIMALS);
+
+			// claim for 2 to move rewards into balance
+			assert_ok!(StakePallet::claim_rewards(Origin::signed(2)));
+			assert!(Balances::free_balance(&2) > DECIMALS);
+			assert!(StakePallet::reward_count(2).is_zero());
+			assert!(StakePallet::rewards(2).is_zero());
+			assert_eq!(Balances::free_balance(&3), DECIMALS);
+
+			// should not be able to claim for incorrect role
+			assert_noop!(
+				StakePallet::increment_collator_rewards(Origin::signed(2)),
+				Error::<Test>::CandidateNotFound
+			);
+			assert_noop!(
+				StakePallet::increment_delegator_rewards(Origin::signed(1)),
+				Error::<Test>::DelegatorNotFound
+			);
+		});
+}
+
+#[test]
+fn api_get_unclaimed_staking_rewards() {
```

Comment:
Can we actually test that the returned value of the function actually matches what is unlocked when calling the extrinsic to pay out those rewards? What do you think?

Reply:
[e703493](https://github.com/KILTprotocol/kilt-node/pull/384/commits/e703493c668c3eb2af734920ac05abca615de5c6)